### ISSUE TITLE
Update needed for pybind11

### ISSUE THREAD

**Issue by subhacom** [BhallaLab/moose-core#463]
On mac with Anaconda, Python 3.10.6, attempt to build moose results in compilation error:
```
... cast.h:446:36: error: member access into incomplete type 'PyFrameObject' (aka '_frame')
                "  " + handle(frame->f_code->co_filename).cast<std::string>() +
                                   ^
/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of '_frame'
typedef struct _frame PyFrameObject;
               ^
```
This is a known issue that has been resolved in newer versions of PyBind11 (https://github.com/pybind/pybind11/discussions/4333?sort=top)

Copying the latest includes from pybind11 into `moose-core/external/pybind11/include/pybind11` solves this issue.

**Comment by hrani** [comment]
Does this still exist? with the latest update hope this is resolved

**Comment by subhacom** [comment]
This should be resolved now.

**Comment by hrani** [comment]
Resolved in commit 8b37d2d691dfa
